src/LiquidityPool.sol

```diff
@@ -300,19 +312,23 @@ contract LiquidityPool is Auth, IERC4626 {
         return share.allowance(owner, spender);
     }
 
-    function transferFrom(address, address, uint256) public returns (bool) {
-        (bool success, bytes memory data) = address(share).call(bytes.concat(msg.data, bytes20(msg.sender)));
+    function transferFrom(address from, address to, uint256 value) public returns (bool) {
+        (bool success, bytes memory data) = address(share).call(
+            bytes.concat(
+                abi.encodeWithSignature("transferFrom(address,address,uint256)", from, to, value), bytes20(msg.sender)
```

Comment:
whats the difference between this and the previous call?

Reply:
I like that its more explicit now, but curious what the reasoning was here.

Reply:
This gets quite deep, but is definitely good to fully understand.

Previously, `transferFrom` was only called externally. Now it is called externally, but also used internally in `requestRedeem`.

Before it used `msg.data + msg.sender` as the full payload. `msg.data` is basically the incoming encoded call. So if you externally call `transferFrom`, it will be `encode(transferFrom(from, to, value))`. But, if you call it from `requestRedeem`, the `msg.data` here will also be `requestRedeem(shares)`. So the call to the tranche token will actually have the request redeem signature, and therefore of course fail.

So instead we are now manually doing the ABI encoding here, rather than using `msg.data`, so it works both for external and internal calls to this function.